src/features/editors/VisualHTMLEditorLine.tsx

```diff
@@ -0,0 +1,57 @@
+import './VisualHTMLEditorLine.css'
+import {Droppable} from "react-beautiful-dnd"
+import Element from "./VisualHTMLEditorElement";
+import {LineSegments, LineSegment} from "./VisualHTMLEditor";
+
+interface ILine {
+  indent: number,
+  id: string,
+  children: LineSegments
+}
+
+function Line(props: ILine) {
+  const makeTag = (tag: LineSegment) => tag.type !== 'text' ?
+    tag.type === 'closing' ? `</${tag.value}>` : `<${tag.value}>` :
+    tag.value + ' '
+
+  function setClasses(element: LineSegment) {
+    let classes: string = element.unlocked ? 'unlocked' : 'locked'
+    if (element.type !== 'text') {
+      classes += element.type === 'opening' ? ' opening' : ' closing'
+    } else {
+      classes += ' text'
+    }
+    return classes
+  }
+
+  return (
+    <Droppable key={props.id} droppableId={props.id} direction={"horizontal"}>
+      {(provided, snapshot) => (
+        <div
+          className={'line'}
+          style={{
+            paddingLeft: 25 * props.indent + 10,
+            backgroundColor: snapshot.isDraggingOver ? '#f0f0f0' : 'white'
```

Comment:
You should set a class for this, for example setting "is-dragging" when this element is being dragged. Then in the CSS you can customize what change in the design occurs when the element is being dragged. This way, you decorrelate the state-driven scripting part (element is being dragged) from the design part (element has background grey).

Reply:
I see, very interesting. I will keep that in mind in the future.
Thank you!